File: src/brs/services/AliasService.java
package brs.services;

import java.util.Collection;

import brs.Alias;
import brs.Alias.Offer;
import brs.Attachment;
import brs.Transaction;
import brs.util.CollectionWithIndex;

public interface AliasService {

  Alias getAlias(long aliasId);

  Alias getAlias(String aliasName, long tld);

  Alias getTLD(String tldName);
  
  Alias getTLD(Long tldId);
  
  Offer getOffer(Alias alias);

  int getAliasCount();

  int getAliasCount(long tld);

  CollectionWithIndex<Alias> getAliasesByOwner(long accountId, String name, Long tld, int from, int to);
  
  CollectionWithIndex<Alias> getTLDs(int from, int to);

  Collection<Alias> getTLDs(long accountId);

  CollectionWithIndex<Alias.Offer> getAliasOffers(long account, long buyer, int from, int to);

  void addOrUpdateAlias(Transaction transaction, Attachment.MessagingAliasAssignment attachment);

  void addTLD(long id, Transaction transaction, Attachment.MessagingTLDAssignment attachment);
  
  void addDefaultTLDs();

  void sellAlias(Transaction transaction, Attachment.MessagingAliasSell attachment);

  void changeOwner(long newOwnerId, Alias alias, int timestamp, boolean updateSubscription);
}

File: src/main/kotlin/brs/services/AliasService.kt
package brs.services

import brs.entity.Alias
import brs.entity.Alias.Offer
import brs.entity.Transaction
import brs.transaction.appendix.Attachment

interface AliasService {
    /**
     * TODO
     */
    fun getAliasCount(): Long

    /**
     * TODO
     */
    fun getAlias(aliasId: Long): Alias?

    /**
     * TODO
     */
    fun getAlias(aliasName: String): Alias?

    /**
     * TODO
     */
    fun getOffer(alias: Alias): Offer?

    /**
     * TODO
     */
    fun getAliasesByOwner(accountId: Long, from: Int, to: Int): Collection<Alias>

    /**
     * TODO
     */
    fun addOrUpdateAlias(transaction: Transaction, attachment: Attachment.MessagingAliasAssignment)

    /**
     * TODO
     */
    fun sellAlias(transaction: Transaction, attachment: Attachment.MessagingAliasSell)

    /**
     * TODO
     */
    fun changeOwner(newOwnerId: Long, aliasName: String, timestamp: Int)
}
